IbisPlugins/GPUVolumeReconstructionAPITest/gpuvolumereconstructionapitestplugininterface.cpp

```diff
@@ -62,15 +62,18 @@ QWidget * GPUVolumeReconstructionAPITestPluginInterface::CreateFloatingWidget()
 
              //Construct ITK Matrix corresponding to VTK Local Matrix
             reconstructor->SetTransform( acq->GetLocalTransform()->GetMatrix() );
-            reconstructor->ReconstructVolume();
+            reconstructor->start();
+            reconstructor->wait();
 
             vtkSmartPointer<ImageObject> reconstructedImage = vtkSmartPointer<ImageObject>::New();
             reconstructedImage->SetItkImage( reconstructor->GetReconstructedImage() );
             QString volName("ReconstructedVolume");
             volName.append( QString::number( i ) );
             reconstructedImage->SetName(volName);
-            sm->AddObject(reconstructedImage.GetPointer(), acq->GetParent()->GetParent() );
-            sm->SetCurrentObject( reconstructedImage.GetPointer() );
+            sm->AddObject(reconstructedImage, sm->GetSceneRoot() );
+            sm->SetCurrentObject( reconstructedImage );
```

Comment:
Why do you need to explicitly create and destroy reconstructor inside the GPU_VolumeReconstruction class. This should be automatic, no?